### PR TITLE
(MAIN-6855) Move adding to the querypage cache to the QueryPage class

### PR DESCRIPTION
**Pull request by Grunny**
Fixes fatal error:

```
PHP Fatal error: Cannot access private property
AllinfoboxesQueryPage::$subpagesBlacklist in /extensions/wikia/
PortableInfobox/PortableInfoboxHooks.class.php on line 166
```

/cc @SebastianMarzjan

### CODE REVIEW

File: extensions/wikia/PortableInfobox/querypage/AllinfoboxesQueryPage.php [hunk 1]
```diff
@@ -96,24 +96,47 @@ public function reallyDoQuery( $limit = false, $offset = false ) {
...
+	}
 
-		return false;
+	protected function hasInfobox( Title $title ) {
+		// omit subages from blacklist
+		return !(
+				$title->isSubpage() &&
+				in_array( mb_strtolower( $title->getSubpageText() ), self::$subpagesBlacklist )
+			) &&
+			!empty( PortableInfoboxDataService::newFromTitle( $title )->getData() );;
```
Comment (sebastianmarzjan) [inline extensions/wikia/PortableInfobox/querypage/AllinfoboxesQueryPage.php]: nitpick: Double colon ;-)

File: extensions/wikia/PortableInfobox/querypage/AllinfoboxesQueryPage.php [hunk 2]
```diff
@@ -96,24 +96,47 @@ public function reallyDoQuery( $limit = false, $offset = false ) {
...
+			] ] )
+			->run( $dbw );
 
-			return !empty( $data );
-		}
+		wfRunHooks( 'AllInfoboxesQueryRecached' );
+	}
 
-		return false;
+	protected function hasInfobox( Title $title ) {
```
Comment (sebastianmarzjan) [inline extensions/wikia/PortableInfobox/querypage/AllinfoboxesQueryPage.php]: What do you think about testing this method with a unit test?

File: extensions/wikia/PortableInfobox/querypage/AllinfoboxesQueryPage.php [hunk 3]
```diff
@@ -96,24 +96,47 @@ public function reallyDoQuery( $limit = false, $offset = false ) {
 				return $out;
 			} );
 
-		return array_filter( $result, [ $this, 'hasInfobox' ] );
+		return array_filter( $result, [ $this, 'filterInfoboxes' ] );
```
Comment (sebastianmarzjan) [inline extensions/wikia/PortableInfobox/querypage/AllinfoboxesQueryPage.php:99]: 👍

File: extensions/wikia/PortableInfobox/PortableInfoboxHooks.class.php [hunk 4]
```diff
@@ -130,41 +130,11 @@ public static function onBacklinksPurge( Array $articles ) {
...
-					$page->getId(),
-					$title->getNamespace(),
-					$title->getDBkey(),
-				] ] )
-				->run( $dbw );
-
-			wfRunHooks( 'AllInfoboxesQueryRecached' );
+		if ( $title->inNamespace( NS_TEMPLATE ) ) {
+			$queryPage = new AllinfoboxesQueryPage();
+			$queryPage->addTitleToCache( $title );
```
Comment (idradm) [inline extensions/wikia/PortableInfobox/PortableInfoboxHooks.class.php]: `$queryPage` assignment could be omitted here

File: extensions/wikia/PortableInfobox/querypage/AllinfoboxesQueryPage.php (hunk 1 above)
Comment (idradm) [inline extensions/wikia/PortableInfobox/querypage/AllinfoboxesQueryPage.php]: Double semicolon there

File: extensions/wikia/PortableInfobox/querypage/AllinfoboxesQueryPage.php (hunk 2 above)
Comment (idradm) [inline extensions/wikia/PortableInfobox/querypage/AllinfoboxesQueryPage.php]: Could it be `private` instead?

File: extensions/wikia/PortableInfobox/querypage/AllinfoboxesQueryPage.php [hunk 5]
```diff
@@ -96,24 +96,47 @@ public function reallyDoQuery( $limit = false, $offset = false ) {
...
+	protected function hasInfobox( Title $title ) {
+		// omit subages from blacklist
+		return !(
+				$title->isSubpage() &&
+				in_array( mb_strtolower( $title->getSubpageText() ), self::$subpagesBlacklist )
+			) &&
+			!empty( PortableInfoboxDataService::newFromTitle( $title )->getData() );;
+	}
+
+	protected function filterInfoboxes( $tmpl ) {
```
Comment (idradm) [inline extensions/wikia/PortableInfobox/querypage/AllinfoboxesQueryPage.php]: Could we make it `private`?

File: extensions/wikia/PortableInfobox/querypage/AllinfoboxesQueryPage.php [hunk 6]
```diff
@@ -96,24 +96,47 @@ public function reallyDoQuery( $limit = false, $offset = false ) {
...
+		$dbw = wfGetDB( DB_MASTER );
+		( new WikiaSQL() )
+			->INSERT()->INTO( 'querycache', [
+				'qc_type',
+				'qc_value',
+				'qc_namespace',
+				'qc_title'
+			] )
+			->VALUES( [ [
+				AllinfoboxesQueryPage::ALL_INFOBOXES_TYPE,
```
Comment (idradm) [inline extensions/wikia/PortableInfobox/querypage/AllinfoboxesQueryPage.php]: Could be `self` now

**Comment by Grunny** [comment]
@SebastianMarzjan @idradm Comments should be addressed. :)

**Comment by idradm** [comment]
LGTM :shipit:

**Pull request merged**